### DISCUSSION TITLE
Questions regarding Updates and Patches

### DISCUSSION THREAD

**Discussion by Fiinleyyy** [oleg-shilo/wixsharp#1614] [Q&A]
Hi,

I'm a little bit confused about updates and patches using WixSharp since there are multiple discussion entries. My question is: What options are available using WixSharp for Updates/Patches. Can I actually create MSP-files or use MajorUpgradeStrategy? And are there any more options?

Thanks in advance

**Comment by oleg-shilo** [comment]
WixSharp supports WiX concept of MajorUpdates but not patches. 
So you cannot build MSP with WixSharp. You will need to use raw wix for that.

The absence of support for patches is deliberate as the current trend in the deployment domain is to rely on a simpler and more reliable modern MajorUpgrade concept. But it's not so uncommon for WiX/MSI to have some features over-engineered :) 

But WiX's `MajorUpgrade` is fully supported. You can find the corresponding sample in the samples library:

![image](https://github.com/user-attachments/assets/78cee275-2ee3-482f-bbaf-1338fdef2160)

`project.MajorUpgrade = ...` is your frieand.

**Comment by Fiinleyyy** [comment]
Ah. Understood. Thank you. MajorUpgrade meaning if I want to update/patch my application I'll backup the necessary data and "re-install" my new application version?

**Comment by oleg-shilo** [comment]
MajorUpgrade in MSI/WiX terminology means that when you install the new version the old one is automatically removed.
Patch (a legacy upgrade strategy) never removes the old deployment but only replaces some files. While it seems like a more flexible approach, it's harder to manage and less reliable.

>  I'll backup the necessary data 

This is an interesting one. 
Good deployment practices require you to avoid installing or even placing the data (e.g. config data, user data) where your product is installed. Instead, you should move it to the user profile. This way you do not need to backup anything on the upgrade.
